src/content/docs/dev/send-tokens/interchain-tokens/integrate-custom-token.mdx

```diff
@@ -39,42 +39,79 @@ contract MyInterchainToken is ERC20, ERC20Burnable {
+
+```solidity
+function registerTokenMetadata(
+    address tokenAddress,  // your token address
+    uint256 gasValue       // gas value
+) external payable {};
+```
+
+This function registers metadata for a token on the [ITS Hub](#). This metadata is used for scaling linked tokens.
+
+> **Note:** Token metadata must be registered before `linkToken` can be called for the corresponding token.
+
+### Register Custom Token with the Interchain Token Factory
+
+After registering token metadata for each of your custom tokens, call the [`registerCustomToken`](https://github.com/axelarnetwork/interchain-token-service/blob/94677f1072da127a0ee7ff5ab6784dcfe11789e8/contracts/InterchainTokenFactory.sol#L515) function on the [Interchain Token Factory](https://github.com/axelarnetwork/interchain-token-service/blob/main/contracts/InterchainTokenFactory.sol).
 
 ```solidity
-bytes32 interchainTokenId = its.deployTokenManager(
-    0xabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdefab, // unique salt for token manager deployment
-    '', // destination chain name (empty string for current chain)
-    ITokenManagerType.TokenManagerType.MINT_BURN, // Token Manager type
-    abi.encode(msg.sender.toBytes(), newTokenProxy), // parameters
-    0 // gas value (zero for local deployment)
-);
+function registerCustomToken(
+    bytes32 salt, // your unique salt value
+    address tokenAddress, // your custom token address
+    TokenManagerType tokenManagerType, // token manager type
+    address operator // minter address
+) external payable returns (bytes32 tokenId) {};
 ```
 
-This function deploys a Token Manager for your token, connects it to ITS upon deployment, and returns the unique interchain token ID.
+This function registers an existing ERC-20 token (your custom token) under a `tokenId` computed from the provided `salt`. A token metadata registration message will also be sent to the ITS Hub. Then, the token can be linked to remote tokens on different chains by submitting the `linkToken` function from the same `msg.sender` using the same `salt`.
 
-## Assign the `minter` Role to Your Token's Token Manager
+### Link Custom Token with the Interchain Token Factory
 
-Transfer the minter role to the Token Manager using the `transferMintership` function:
+To link a custom token, use the [`linkToken`](https://github.com/axelarnetwork/interchain-token-service/blob/94677f1072da127a0ee7ff5ab6784dcfe11789e8/contracts/InterchainTokenFactory.sol#L541) function to link your custom tokens on your respective chains.
 
 ```solidity
-myToken.transferMintership(tokenManagerAddress);
+function linkToken(
+    bytes32 salt, // your unique salt value previously used
+    string calldata destinationChain, // destination chain
+    bytes calldata destinationTokenAddress, // destination custom token address
+    TokenManagerType tokenManagerType, // token manager type
+    bytes calldata linkParams, // minter address
```

Comment:
I think this is the operator address not necessarily the minter address

Reply:
updated